packages/fetchai/skills/generic_buyer/handlers.py

```diff
@@ -177,8 +177,10 @@ def _handle_decline(
         )
         target_message = fipa_dialogue.get_message_by_id(fipa_msg.target)
 
+        # to tell mypy target isn't none (it cannot be because this makes it
+        # invalid, therefore rejected by the dialogue when trying to add it)
```

Comment:
Please remove; that will confuse people more than help imo.

Reply:
I'll remove these. 

I was very frustrated with this stupid check that is totally unnecessary because we went to all the trouble to design it in a way that all dialogue specific checks are on one place, and then you have to add this stupid check in your code and mess it up just because mypy can't understand the value cannot be none >:(

Reply:
I'll remove these

Reply:
But the method doesn't know about the id you're passing to it. For all the method knows it is accepting an int and trying to find a message for it.